include/deal.II/base/subscriptor.h

```diff
@@ -104,21 +104,44 @@ class Subscriptor
 
   /**
    * Subscribes a user of the object by storing the pointer @p validity. The
-   * subscriber may be identified by text supplied as @p identifier.
+   * subscriber may be identified by text supplied as @p identifier. The latter
+   * variable must be not be a temporary and its type must decay to
+   * const char *. In particular, calling this function with a rvalue reference
+   * is not allowed.
+   */
+  template <typename ConstCharStar = const char *>
+  typename std::enable_if<
+    std::is_same<ConstCharStar, const char *>::value>::type
+  subscribe(std::atomic<bool> *const validity,
+            ConstCharStar            identifier = nullptr) const;
+
+  /**
+   * Calling subscribe() with a rvalue reference as identifier is not allowed.
    */
   void
   subscribe(std::atomic<bool> *const validity,
-            const char *             identifier = nullptr) const;
+            const char *&&           identifier) const = delete;
 
   /**
    * Unsubscribes a user from the object.
    *
    * @note The @p identifier and the @p validity pointer must be the same as
-   * the one supplied to subscribe().
+   * the one supplied to subscribe(). The latter variable must be not be a
+   * temporary and it stype must decay to const char *. In particular, calling
+   * this function with a rvalue reference is not allowed.
+   */
+  template <typename ConstCharStar = const char *>
+  typename std::enable_if<
+    std::is_same<ConstCharStar, const char *>::value>::type
+  unsubscribe(std::atomic<bool> *const validity,
+              ConstCharStar            identifier = nullptr) const;
```

Comment:
But here isn't it the case that you want to compare the string pointed to with one you had previously stored? Why couldn't you pass in an revalue reference here? You're not storing `identifier` for later use, are you?

Reply:
Yes, correct. The idea here is to have a symmetric interface as for subscribe.
Arguably, the chances are higher that the memory address passed as rvalue reference here t is the one intended are higher. If you want I can revert the change to `unsubscribe`.